File: Quiz PBO/GUI Form Kereta/Kereta.py
import requests
import json
class Kereta:
    def __init__(self):
        self.__id=None
        self.__id_kereta = None
        self.__tujuan = None
        self.__id_gerbong = None
        self.__url = "http://localhost/appkereta/kereta_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def id_kereta(self):
        return self.__id_kereta
        
    @id_kereta.setter
    def id_kereta(self, value):
        self.__id_kereta = value
    @property
    def tujuan(self):
        return self.__tujuan
        
    @tujuan.setter
    def tujuan(self, value):
        self.__tujuan = value
    @property
    def id_gerbong(self):
        return self.__id_gerbong
        
    @id_gerbong.setter
    def id_gerbong(self, value):
        self.__id_gerbong = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_id_kereta(self, id_kereta):
        url = self.__url+"?id_kereta="+id_kereta
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id']
            self.__id_kereta = item['id_kereta']
            self.__tujuan = item['tujuan']
            self.__id_gerbong = item['id_gerbong']
        return data
    def simpan(self):
        payload = {
            "id_kereta":self.__id_kereta,
            "tujuan":self.__tujuan,
            "id_gerbong":self.__id_gerbong
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_id_kereta(self, id_kereta):
        url = self.__url+"?id_kereta="+id_kereta
        payload = {
            "id_kereta":self.__id_kereta,
            "tujuan":self.__tujuan,
            "id_gerbong":self.__id_gerbong
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_id_kereta(self,id_kereta):
        url = self.__url+"?id_kereta="+id_kereta
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text

File: Quiz PBO/App API/App Kereta/Kereta.php
<?php
require_once 'database.php';
class Kereta 
{
    private $db;
    private $table = 'kereta';
    public $id_kereta = "";
    public $tujuan = "";
    public $id_gerbong = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_id_kereta(int $id_kereta)
    {
        $query = "SELECT * FROM $this->table WHERE id_kereta = $id_kereta";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`id_kereta`,`tujuan`,`id_gerbong`) VALUES ('$this->id_kereta','$this->tujuan','$this->id_gerbong')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET id_kereta = '$this->id_kereta', tujuan = '$this->tujuan', id_gerbong = '$this->id_gerbong' 
        WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_id_kereta($id_kereta): int
    {
        $query = "UPDATE $this->table SET id_kereta = '$this->id_kereta', tujuan = '$this->tujuan', id_gerbong = '$this->id_gerbong' 
        WHERE id_kereta = $id_kereta";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_id_kereta($id_kereta): int
    {
        $query = "DELETE FROM $this->table WHERE id_kereta = $id_kereta";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>